cf/src/main/scala/com/timeout/scalacloudformation/CfExp.scala

```diff
@@ -1,22 +1,30 @@
 package com.timeout.scalacloudformation
 
-import java.time.ZonedDateTime
+import java.time.{Duration, ZonedDateTime}
 
-import com.timeout.scalacloudformation.AWSResources.{CfExp, Resource}
 import io.circe.Json
 
+trait CfExp[+T]
+
 object CfExp {
   type E[+T] = CfExp[T]
-  sealed trait Lit[+T] extends E[T] {
-    def value: T
+
+  trait IsLit[A]
+
+  object IsLit {
+    implicit val stringLit: IsLit[String] = new IsLit[String] {}
+    implicit val IntLit: IsLit[Int] = new IsLit[Int] {}
+    implicit val longLit: IsLit[Long] = new IsLit[Long] {}
+    implicit val doubleLit: IsLit[Double] = new IsLit[Double] {}
+    implicit val boolLit: IsLit[Boolean] = new IsLit[Boolean] {}
+    implicit val dateTimeLit: IsLit[ZonedDateTime] = new IsLit[ZonedDateTime] {}
+    implicit val jsonLit: IsLit[Json] = new IsLit[Json] {}
+    implicit val durationLit: IsLit[Duration] = new IsLit[Duration] {}
+    implicit def propertyLit[T <: ResourceProperty]: IsLit[T] = new IsLit[T] {}
+    implicit def listLit[A: IsLit]: IsLit[List[A]] = new IsLit[List[A]]{}
```

Comment:
Only lists of strings are permitted according to their spec
http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resources-section-structure.html